### ISSUE TITLE
Request `print` support for dictionaries; and should not throw `LCompilersException`

### ISSUE THREAD

**Issue by rebcabin** [lcompilers/lpython#1971]
Two issues: (1) `LPython should produce an error message and not a traceback (see bottom):

(2) Dict print is really useful for `print` debugging, e.g., 

```python
def lexer_test():
    print('LEXER TEST')
    lexer : LasrLexer = LasrLexer(StringIO('  foo  bar  '))
    eat_white(lexer)
    t : str = peek(lexer)
    print({"t": t})
    assert t == 'f'
```

produces today:

```bash
/Users/brian/CLionProjects/lpython/src/bin/python /Users/brian/CLionProjects/lpython/lasr/LP-pycharm/lasr_lexer.py 
Internal Compiler Error: Unhandled exception
Traceback (most recent call last):
  File "/Users/brian/CLionProjects/lpython/src/bin/lpython.cpp", line 1844
    err = compile_python_to_object_file(arg_file, tmp_o, runtime_library_dir,
  File "/Users/brian/CLionProjects/lpython/src/bin/lpython.cpp", line 809
    res = fe.get_llvm3(*asr, pass_manager, diagnostics, infile);
  File "/Users/brian/CLionProjects/lpython/src/lpython/python_evaluator.cpp", line 58
    run_fn, infile);
  File "/Users/brian/CLionProjects/lpython/src/libasr/codegen/asr_to_llvm.cpp", line 8518
    v.visit_asr((ASR::asr_t&)asr);
  File "../libasr/asr.h", line 4869
  File "../libasr/asr.h", line 4845
  File "../libasr/asr.h", line 4870
  File "../libasr/asr.h", line 4585
  File "/Users/brian/CLionProjects/lpython/src/libasr/codegen/asr_to_llvm.cpp", line 1359
    ASR::symbol_t *mod = x.m_global_scope->get_symbol(item);
  File "../libasr/asr.h", line 4872
  File "../libasr/asr.h", line 4593
  File "/Users/brian/CLionProjects/lpython/src/libasr/codegen/asr_to_llvm.cpp", line 2925
    finish_module_init_function_prototype(x);
  File "/Users/brian/CLionProjects/lpython/src/libasr/codegen/asr_to_llvm.cpp", line 4321
    visit_Function(*s);
  File "/Users/brian/CLionProjects/lpython/src/libasr/codegen/asr_to_llvm.cpp", line 3926
    visit_procedures(x);
  File "/Users/brian/CLionProjects/lpython/src/libasr/codegen/asr_to_llvm.cpp", line 4272
    this->visit_stmt(*x.m_body[i]);
  File "../libasr/asr.h", line 4889
  File "../libasr/asr.h", line 4631
  File "/Users/brian/CLionProjects/lpython/src/libasr/codegen/asr_to_llvm.cpp", line 7134
    handle_print(x);
  File "/Users/brian/CLionProjects/lpython/src/libasr/codegen/asr_to_llvm.cpp", line 7333
    throw LCompilersException("Printing support is not available for " +
LCompilersException: Printing support is not available for dict type.```

**Comment by certik** [comment]
Yes, it should do CodeGenError, which shows a nice error message.

And yes, we should implement it.

**Comment by arteevraina** [comment]
> Two issues: (1) `LPython should produce an error message and not a traceback (see bottom):
> 
> (2) Dict print is really useful for `print` debugging, e.g.,
> 
> ```python
> def lexer_test():
>     print('LEXER TEST')
>     lexer : LasrLexer = LasrLexer(StringIO('  foo  bar  '))
>     eat_white(lexer)
>     t : str = peek(lexer)
>     print({"t": t})  ############# ATTENTION ##############
>     assert t == 'f'
> ```
> 

In my local environment, ``LasrLexer`` is not being identified by the compiler.

```
(lp) arteevraina@Arteevs-MacBook-Air lpython % src/bin/lpython src/a.py                
semantic error: Unsupported type annotation: LasrLexer
  --> src/a.py:12:13
   |
12 |     lexer : LasrLexer = LasrLexer(StringIO('  foo  bar  '))
   |             ^^^^^^^^^ 
```

**Comment by rebcabin** [comment]
Hi Arte ... yes, I forgot to push the repro case for this one. I'll either fix that today or close the issue as incomplete. My apologies!

**Comment by arteevraina** [comment]
> Hi Arte ... yes, I forgot to push the repro case for this one. I'll either fix that today or close the issue as incomplete. My apologies!

Thank you for your reply @rebcabin .

I am able to reproduce this issue with an easier example.

```
def lexer_test():
    print({"t": "t"})

lexer_test()
```

```
(lp) arteevraina@Arteevs-MacBook-Air lpython % src/bin/lpython src/a.py
Internal Compiler Error: Unhandled exception
Traceback (most recent call last):
  File "/Users/arteevraina/GSI/lpython/src/bin/lpython.cpp", line 1844
    err = compile_python_to_object_file(arg_file, tmp_o, runtime_library_dir,
  File "/Users/arteevraina/GSI/lpython/src/bin/lpython.cpp", line 809
    res = fe.get_llvm3(*asr, pass_manager, diagnostics, infile);
  File "/Users/arteevraina/GSI/lpython/src/lpython/python_evaluator.cpp", line 58
    run_fn, infile);
  File "/Users/arteevraina/GSI/lpython/src/libasr/codegen/asr_to_llvm.cpp", line 8538
    v.visit_asr((ASR::asr_t&)asr);
  File "../libasr/asr.h", line 4869
  File "../libasr/asr.h", line 4845
  File "../libasr/asr.h", line 4870
  File "../libasr/asr.h", line 4585
  File "/Users/arteevraina/GSI/lpython/src/libasr/codegen/asr_to_llvm.cpp", line 1359
    ASR::symbol_t *mod = x.m_global_scope->get_symbol(item);
  File "../libasr/asr.h", line 4872
  File "../libasr/asr.h", line 4593
  File "/Users/arteevraina/GSI/lpython/src/libasr/codegen/asr_to_llvm.cpp", line 2940
    finish_module_init_function_prototype(x);
  File "/Users/arteevraina/GSI/lpython/src/libasr/codegen/asr_to_llvm.cpp", line 4336
    visit_Function(*s);
  File "/Users/arteevraina/GSI/lpython/src/libasr/codegen/asr_to_llvm.cpp", line 3941
    visit_procedures(x);
  File "/Users/arteevraina/GSI/lpython/src/libasr/codegen/asr_to_llvm.cpp", line 4287
    this->visit_stmt(*x.m_body[i]);
  File "../libasr/asr.h", line 4889
  File "../libasr/asr.h", line 4631
  File "/Users/arteevraina/GSI/lpython/src/libasr/codegen/asr_to_llvm.cpp", line 7154
    handle_print(x);
  File "/Users/arteevraina/GSI/lpython/src/libasr/codegen/asr_to_llvm.cpp", line 7353
    throw LCompilersException("Printing support is not available for " +
LCompilersException: Printing support is not available for dict type.
```

**Comment by rebcabin** [comment]
@arteevraina  here is the repro case :)

https://github.com/rebcabin/lpython/tree/brian-lasr/lasr/LP-pycharm/Issue1971
